### DISCUSSION TITLE
Access Point Questions

### DISCUSSION THREAD

**Discussion by tmqdeborah** [crystalyragui/MARC2RDA#453] [General]
### Here are questions we need to answer related to access points
******* Please reply to each question separately, using "Write a reply" instead of starting a new comment. *******

**Comment by tmqdeborah** [comment]
### Question: What should we use as internal punctuation for access points?

N.B. Please reply to each question separately, using the"Write a reply" box.

**Reply by cspayne** [reply]
I think we should retain whatever internal punctuation is provided, without referring to MARC coding for cataloging rules applied. - Deborah

**Reply by tmqdeborah** [reply]
If we are, instead, going to 'fix' old or erroneous punctuation, we will need to set up a set of conditions and punctuation rules. And this could get very complicated, very quickly.

**Reply by GordonDunsire** [reply]
I think there's a bit of overloading here. We should use the punctuation that is provided, and make no judgements about how 'correct' it is. We should not attempt to fix anything. If no punctuation is provided, we should insert the same delimiter between components of the heading, for example semi-colon-space or space-hyphen-space. All of this is about strings, which are of secondary consideration in a linked data application.

**Reply by cspayne** [reply]
DECIDED: We will retain the internal punctuation provided.

**Comment by cspayne** [comment]
### Question: What ending punctuation do we want to keep when putting together access points?

Because we're concatenating subfields to produce access point strings (see [Access Point Mapping Table](https://docs.google.com/spreadsheets/d/172rsUQOJLk8dL71r4KweQr6ZUf2RULFc/edit#gid=673207336)), these strings may end in punctuation that we don't want. Can we create a list of ending punctuation we do not want to keep so that the transform can remove these? 

For example,
```xml
<marc:datafield tag="100" ind1="1" ind2=" ">
       <marc:subfield code="a">Boulanger, John,</marc:subfield>
       <marc:subfield code="d">1965-</marc:subfield>
</marc:datafield>
```
would result in the string "Boulanger, John, 1965-", where we want to keep the hyphen. 

```xml
 <marc:datafield tag="700" ind1="1" ind2=" ">
      <marc:subfield code="a">Williams, Judy</marc:subfield>
      <marc:subfield code="c">(Wildlife manager),</marc:subfield>
      <marc:subfield code="e">author.</marc:subfield>
</marc:datafield>
```
would result in "Williams, Judy (Wildlife manager)," and we may not want to keep that ending comma.

**Reply by GordonDunsire** [reply]
Any punctuation appearing at the end of ap/aap string should be removed. That is, any character that is not a-z/A-Z. And the process should be iterative until the last character in the string is a-z/A-Z.

**Reply by cspayne** [reply]
Removing _all_ ending punctuation will result in access points like "Williams, Judy (Wildlife manager" and "Randall, Margaret, 1936" for a $d that was "1956-", are those not worth keeping?

**Reply by tmqdeborah** [reply]
Removing ending punctuation that is a period is also problematic when a headings field ends in an initial or an abbreviation, etc, e.g.:

- 100 1  $a Blair, Frank P.
- 100 1  $a Hopkins, Griffith Morgan, $c Jr.

**Reply by cspayne** [reply]
Reply from Laura via email (moved to this thread to keep discussion organized): 

@ cypress I think there are tools for this, I just can't put my finger on them at the moment.  it had to do with a PCC decision to stop supporting ISBD punctuation in cataloging.  Here's more:  Punctuate the Positive, or Eliminate the Negative with notes.pdf (librarylearning.org)<https://librarylearning.org/system/files/2020-10/Punctuate%20the%20Positive%2C%20or%20Eliminate%20the%20Negative%20with%20notes.pdf>   If you put a message on Code4Lib you might get a response from someone familiar with code that depunctuates MARC fields, and whatever it's written in, that might help you.  I just don't have time to look today.

**Reply by GordonDunsire** [reply]
I'm just trying to keep it as simple as possible :-) I think we can live with "Jr", so what else is there other than hyphen, end parenthesis, and space-A-Z-stop (which is easy for a regex function)?

**Reply by tmqdeborah** [reply]
According to the research that I did for my book "Cataloging with AACR2 and MARC21" regarding End of field punctuation" (for common fields at the time of writing--2006) The EOF punc for 1XX, 6XX, 7XX, and 8XX can be a: Period '.' question mark '?' exclamation mark'!' hyphen '-' or parenthesis ')'

Examples:
110 2  $a C.M. Bell (Firm : Washington, D.C.)
100 1  $a Johnson, T. $q (Thomas) $c (Engraver)
100 1  $a Powers, Tom $q (Tom J.)
100 1  $a Durrett, Deanne, $d 1940-
100 1  $a Dunbar, William, $d 1460?-1520?
130 0  $a Microsoft Plus!

I was almost coming around to agreeing that we should:
- remove ending punctuation marks, including periods, except after initials (because (as Gordon says) they can be retained whenever the heading string ends with a 'space[A-Z].')
- keep only: question mark '?' exclamation mark'!' hyphen '-' or parenthesis ')' 

But there are quite a few **abbreviations** that will look very peculiar without the ending period:
110 2  $a Sarony & Co.
100 1  $a Ball, Michael, $c Revd. Dr.
100 1  $a Wesseling, Pieter, $c Dr. Ir.
100 1  $a Poling, Jim, $c Sr.
110 1  $a New Jersey. $b Commissioners on Fourteenth Regiment Monument at Monocacy, Md.
110 2  $a Woman's Institute of Domestic Arts and Sciences. $g Scranton, Pa.
100 1  $a Davidson, Dorothy, $c B.Sc.
100 1  $a Taylor, D. A., $c M.Sc.
100 1  $a Ware, Leonard, $c Lt.

I now think it will be worse to give up on periods for abbreviations than to just all periods at the end of an access point. And, in fact, maybe it would be better to **keep all punctuation at the end of an access point except the comma ','** just in case there is a perfectly valid double-quotation mark ( " ) at the end of a name or title, etc.

It might look strange to have the ending period, but it will be better (I think) than losing that it indicates an abbreviation.

I have attached a PDF of my EOF table, just FYI.
[EOF punctuation.pdf](https://github.com/user-attachments/files/15775128/EOF.punctuation.pdf)

**Reply by CECSpecialistI** [reply]
I think we need to keep the ending periods so we don't mess up the abbreviations.

**Reply by lake44me** [reply]
Hi @tmqdeborah. @cspayne and I have been talking about using a regular expression to avoid stripping the ending period from abbreviations.  It could look for a situation where the last word before the period is a single letter or two letters.  Your examples with degrees, e.g. "M.Sc." is a wrinkle - but only if definition of "word" has either a space or a period in front of it.   If we can spot other examples where an ending period is part of the text, that don't fit this pattern but fit another one, it could help refine this process.  If we could get this to work, say, 99% of the time, would that be worth using it?

**Reply by lake44me** [reply]
I'm realizing that normalizing the 245 title data is a different case than normalizing access points in 1xx 6xx 7xx 8xx etc. .  For the 245, I think only the ending period should be attempted if we can avoid periods after initials and abbreviations.  For names, it probably isn't necessary to try to strip periods, but can we get rid of forward angle brackets and commas?

**Reply by cspayne** [reply]
For now, the code will retain all ending punctuation apart from commas for agent access points. When we take a look at output data from the transform, we will be able to see if there are other punctuation we would like to handle differently.

**Comment by tmqdeborah** [comment]
### Question: Is it ok to map all valid name subfields from a heading for an Agent entity, even if some of those subfields are not typically used?
For example, a subfield $g is not usually used for a Personal name, but here is a real example:
700 1  $a Meredov, N. A. $g (Nepesmurad Aksakalovich)  

Note that:

- If a valid subfield name subfield is used erroneously, as in the example above, the cataloger still intended it to be a qualifier, and it will still function as a qualifier in an access point. e.g.:
         Meredov, N. A. (Nepesmurad Aksakalovich)

- of course, there will be some 'real' errors, e.g.:
     700 1  $a Tanner, Bruce, $g ill.      = Tanner, Bruce, ill.

**Reply by cspayne** [reply]
When discussed in our optional meeting on June 4, 2024 there were no objections from the group to doing so.

**Reply by GordonDunsire** [reply]
We should be focusing on the actual data values, not what is correct or typically used. If a subfield is not used in a data value, so what?

**Reply by CECSpecialistI** [reply]
"Not typical" isn't the same as "not allowed" or "incorrect", and I don't know if we should be tossing out values if they aren't flat-out invalid MARC. Or if we have time to do so. Just my two cents.

**Reply by GordonDunsire** [reply]
I think we remain in agreement; we are arguing for the retention of values that appear in the record :-)

**Reply by lake44me** [reply]
FWIW, I agree.  Apparently subfield g in 100 / 700 gets used for family names to identify the family by one of its members as a qualifier, and would thus be part of the authorized heading.  [https://www.loc.gov/aba/rda/mgd/family/mg-f-prominentMemberOfFamily-01.pdf](chrome-extension://efaidnbmnnnibpcajpcglclefindmkaj/https://www.loc.gov/aba/rda/mgd/family/mg-f-prominentMemberOfFamily-01.pdf)

**Reply by cspayne** [reply]
DECIDED: We are in agreement to use all valid name subfields in headings for Agent access points, even when they are not typically used.

**Comment by tmqdeborah** [comment]
### Question: Is it ok to transform a heading in a 1XX, 6XX, 7XX, or 8XX as an Authorized access point for an entity when no $0 or $1 has been provided?

- According to MARC Bib “Content designators identify the subelements occurring in personal name fields constructed according to generally accepted cataloging and thesaurus-building rules (e.g., Anglo-American Cataloguing Rules (AACR 2), Library of Congress Subject Headings (LCSH). -- https://www.loc.gov/marc/bibliographic/bdx00.html
  - This means that these headings are “controlled”, i.e., structured as per, e.g., the [second option](https://access.rdatoolkit.org/en-US_ala-5ca9e124-575a-3e04-aec3-b04c6a482349/div_yky_kbw_qhb) for recording an **Access** point for person “Construct an access point by applying a string encoding scheme to the values of one or more other elements.”
  - [This option](https://access.rdatoolkit.org/en-US_ala-335fd204-e6f8-39ed-8681-d56c985fab59/div_lmd_kbw_qhb) (to apply an SES) is also provided for recording a structured description of an **Authorized** access point for person.
- And, as Damien Iseminger has said in a recent NACO related email, “Any heading that is in the LC Catalog will also be seen in authorities.loc.gov. Unlike today, an authority was not created in every instance for a new name when an item was cataloged. If there was no need for a cross-reference, the main entry card would suffice as documentation of the authorized form of heading”. 
  - This means that headings that are not in the LC catalog, but are constructed according to the same SES that is used by LC, should be considered to be authorized, as Adam Schiff said in the June 6, 2024 meeting. 

  - **Question**: Is there a MARC code that indicates NACO conformance, if it turns out that we need it? 

- It is definitely not safe to use an **Access** point as the local portion of an IRI (see Minting IRIs). So, if we record a heading as an Access point instead of an **Authorized** access point for an entity, then we will not be able to prevent proliferation of IRIs for the same entity for a large proportion of the headings in legacy files.

For these reasons, my reply to this questions is "Yes, it is ok to transform a heading in a 1XX, 6XX, 7XX, or 8XX as an Authorized access point for an entity when no $0 or $1 has been provided."

**Reply by GordonDunsire** [reply]
If there is no information about the source of a heading, the default assumption must be that it is an access point (a string label that is structured according to some scheme). Also, the method of using the heading as the local part of a minted IRI for the entity is not valid. Instead, a running number or similar must be used. If there is information the source, the source code must be used as the super-folder of a minted IRI for the entity, or concatenated with the local part; e.g. ... /person/lcsh/concatenatedHeading or .../person/lcshConcatenatedHeading. I prefer the latter.

In other words, if there is a source, the minted IRI for the entity includes the source, the local part is based on the source+heading, and the minted nomen is linked to its entity as an aap. The minted nomen is linked to the source.; If there is no source, the minted IRI does not include a source, the local part is a running number, there is no justification for minting a nomen, and the heading string is linked to the entity as an access point.

There is nothing we can or should do to fix "authority" data if no authority is recorded.

**Reply by tmqdeborah** [reply]
Why would RDA provide the **AAP** [option ](https://access.rdatoolkit.org/en-US_ala-335fd204-e6f8-39ed-8681-d56c985fab59/div_lmd_kbw_qhb) to “Construct an access point by applying a string encoding scheme to the values of one or more other elements.” if that option did not apply to an AAP?

An authorized access point is an "access point access point for ... that is selected for preference in a specific vocabulary encoding scheme". (see [AAP for person](https://access.rdatoolkit.org/en-US_ala-335fd204-e6f8-39ed-8681-d56c985fab59/c6500282-840b-4a1f-9fd1-25d8fc602384))

According to the [MARC standards](https://www.loc.gov/marc/bibliographic/bdx00.html), "The subelements occurring in personal name fields constructed according to generally accepted cataloging and thesaurus-building rules (e.g., Anglo-American Cataloguing Rules (AACR 2), Library of Congress Subject Headings (LCSH))", i.e., they are (supposed to be) constructed according to an SES.

Just as it is optional in RDA to record the VES or SES of an AAP as data provenance, it is optional in MARC to record the VES or SES of a heading. But, given the MARC instruction to construct a heading according to an SES, why can't we apply the AAP SES option and say that even if we do not know that a heading is recorded in a specific VES, we know that it is (supposed to be) constructed according to an SES and is, therefore, an access point that has been selected for preference? At least for LC records and BL and other national library records?

This is not an idle question. It matters whether we map these headings without source data as AAP or AP.

We still have to discuss and make final decisions about our methods for [minting IRIs](https://github.com/uwlib-cams/MARC2RDA/discussions/450)--re. using a heading as the local part of an IRI when possible vs always using opaque numbers--but I agree with Gordon that, in our transform, "the method of using the heading as the local part of a minted IRI for the entity" should not be used for an Access Point; it should only be used for an Authorized Access Point. 

But, since the vast majority of all types of headings in MARC records (except subject headings) do not have 'source of heading' data, if we cannot record them as AAPs we will not be able to use them to prevent proliferation of minted entities. 

I suggest that, for our transform, our mapping should be: if there is no source, the minted IRI for the entity does not include a source, the local part is based on the AAP heading alone, the minted nomen is linked to its entity as an aap, and the scheme of nomen is "transform" as a warning that a heading should be checked against an authority file at some point.

**Reply by GordonDunsire** [reply]
We are only minting a related entity when a) we can attach an appellation for conformance with RDA and b) when we can extract additional descriptive data from the record and attach it. If these conditions are not met, we should not mint a related entity and just use a literal for the record's data value.

We have agreed, I think, that if a source is given for a heading for a related entity but an IRI is not recorded, that is sufficient to meet the condition for minting our own IRI and attach the heading as an appellation. We expect the given source to be processed as data provenance, as the nomen scheme of the nomen string that is the appellation/heading. This requires the appellation to be minted as a nomen.

We have also agreed, I think, that a nomen is only minted if there is a source for the heading/nomen string.

If an IRI for the related entity/rwo is given, in many cases the de-referenced description set includes the heading as a string, so there seems to be limited cost-benefit in minting a heading nomen, etc. However, we have agreed, I think, to mint nomens for all subject headings, etc. because the coverage in the source schemes is patchy or messy.

If no source is given for a heading, we cannot assume what the source is, and therefore which SES is used. We can assume that the appellation/heading is an access point because it is structured in some unknown way and it is clearly not an identifier. We cannot assume that the heading is authorized for two reasons.

1. No authorization process has taken place and no attempt has been made to disambiguate the heading. This applies to significant quantities of records, for example publishers' metadata.

2. The authorization is local, and the heading has only been disambiguated against the local copies of authority and dataset files. This covers cases of local disagreement/user-focus with national norms, for example the addition of death dates to already disambiguated access points for persons.

If the heading is not disambiguated in a known scheme, the 'proliferation of minted entities' has to be balanced against the impact of erroneously merging two distinct entities. In previous discussions we have agreed, I think, that it is better to say too much than to say the wrong thing :-)

I think the suggestion of 'transform' as a nomen scheme is actually more work and less safe than just using the heading as a literal access point for a non-disambiguated related entity. And both require transform warnings.

**Reply by CECSpecialistI** [reply]
DECISION

**Reply by CECSpecialistI** [reply]
Follow-up: We will attempt to do pre-processing on our data to reconcile as many access points as possible against the LCNAF

**Reply by tmqdeborah** [reply]
Here is my understanding of what was agreed in the meeting today:
- Run pre-processing on the data to [reconcile as many access points as possible against the LCNAF](https://github.com/uwlib-cams/MARC2RDA/discussions/453#discussioncomment-9885015) 
- If, after this pre-processing, a headings field for any entity (including Agents as well as Works and Expressions) does not have an approved $0 or $1 or $2 that provides a source VES for the heading, then:
  - Mint an opaque IRI for the entity
  - Record the appropriate access point subfields from the heading as an 'access point for [entity]' (not an authorized access point)
  - Do not mint a nomen for the access point

Please correct anything I might have missed or misunderstood.

**Comment by tmqdeborah** [comment]
### Question: Should we try to standardize the format for family names? 
See the examples of family names from approx 5.6 million LC records from 2022: [100I1-3_700I1-3_LC.
](https://drive.google.com/drive/folders/1m5tP46eD2E00T5RBQ5qfOwzeDXvRANAn
)

The consensus at the meeting on July 6, 2024 was that we should not try to fix data and should just map the data as we find it.

**Reply by GordonDunsire** [reply]
Apart from the principle, I see no programmatic way of fixing this mess :-)

**Reply by cspayne** [reply]
DECIDED: We will not try to fix data and should just map the data as we find it.

**Comment by tmqdeborah** [comment]
### Question: What do we add as a language qualifier for an AP for expression if more than one language code is provided in 041$a?

If we are using a language of expression as a qualifier for an access point for expression, and there are no expression qualifiers in a main entry field, we can [map the language code from 008/35-37](https://id.loc.gov/vocabulary/languages.html). But what do we do when there is more than one language code in [041](https://www.loc.gov/marc/bibliographic/bd041.html)$a?

Under [Additional elements and designations in access points for expression](https://access.rdatoolkit.org/en-US_ala-0eb21eb3-a2b4-3f0c-b76f-7df760946aad/p_mwk_1nw_5fb) RDA offers the option to: “Include a value of Expression: [language of expression](https://access.rdatoolkit.org/Content/Index?externalId=en-US_ala-b7de61e2-fe4b-3cc3-a4a2-7a3c4c55b4fc).
It does not say whether or not we can include more than one value of expression

@GordonDunsire, is there any reason that we cannot include more than one value of expression in situations such as the below—whether the AP is for a single work with multiple languages or a single work in multiple languages?

008 000413s2000    enk           000 0 eng  
010    $a    00031058 
041 0  $a eng $a rus
100 1  $a Murray, John 
245 10 $a Intermediate Russian : $b a grammar and workbook / $c John Murray and Sarah Smyth.
- Expr. AP: Murray, John. Intermediate Russian. English. Russian

008 000920s2001    mdu      b    001 0 eng  
010    $a    00048817 
041 0  $a eng $a heb
100 1  $a Zimmermann, Frank.
245 10 $a Before the masora / $c Frank Zimmermann.
546    $a Text in English with some Hebrew.
- Expr. AP: Zimmermann, Frank. Before the masora. English. Hebrew

008 000719s2000    gw a          000 0 eng  
010    $a    00059985 
041 0  $a eng $a fre $a ger
245 00 $a Educating for sustainability = $b Umweltbildung und Agenda 21 = Former a la durabilitâe / $c Gerhard de Haan, Julia Mann, Anna Marie Reid (Hrsg./eds.).
546    $a English, French, and German.
- Expr. AP: Educating for sustainability = Umweltbildung und Agenda 21 = Former a la durabilitâe. English. French. German

**Reply by tmqdeborah** [reply]
Verbal reply from @GordonDunsire during the Transform meeting today, saying (basically) that RDA does not prohibit recording multiple languages as qualifiers for an Expression. That is left to community resources. And we can say that our community (the UW M2R transform project) allows recording multiple languages as qualifiers for an Expression.

So, I have added the below to the "Single Expr AP" tab in the [Access Point Mapping Table.20230814](https://docs.google.com/spreadsheets/d/1CXBDgFvXzFiRytEz8G-9dkax8oyEA3q_/edit?gid=454205194#gid=454205194):
Find language from 008/35-37: Extend this to 
- IF 041 contains multiple language codes in separate subfields $a (e.g., $a eng $e fre $e ger) or strung in a single subfield (e.g., $a engfreger), then convert all $a codes into language terms from [MARC List for Languages](https://id.loc.gov/vocabulary/languages.html) record them in a string separated by periods after the title or name/title values, preceded by a period
  - Access point for expression: Zimmermann, Frank. Before the masora. English. Hebrew. 
- ELSE IF multiple language codes are not present in 041$a, AND a language code is present in 008/35-37, then convert the code into its mapped term and record that term after title or name/title values, preceded by a period
  - Access point for expression: Kerner, Charlotte, Blueprint. English 
- ELSE IF a language code is not present in 008/35-37 AND a single language code is present in 041$a, then convert the code into its mapped term and record that term after title or name/title values, preceded by a period
